### PR TITLE
Ft cart 187300163

### PR DESCRIPTION
**Pull request by RWEMAREMY**
**What does this PR do?**
This PR implements the functionality of Cart for Buyer 

**Description of Task to be completed?**
This task involves adding Cart functionality and listing all products at the app's backend,

**How should this be manually tested?**
Run npm install to install all dependencies.
Start the development server with npm run dev.
Use swagger for testing
Any background context you want to provide?
This feature enhances the app's user experience by providing a convenient and effective usability to utelize Cart  through purchasing various product. 

**What are the relevant pivotal tracker/Trello stories?**
Story ID:[#187300163]

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -111,7 +111,8 @@
     "src/config/config.js",
     "test/*/.ts",
     "swagger.json",
-    "src/database/models/role.ts"
+    "src/database/models/role.ts",
+    "src/database/models/cart.js"
```
Comment (fab-ryan) [inline tsconfig.json]: why this file??

**Review by fab-ryan — CHANGES_REQUESTED** [review]
work on those also

File: src/controllers/cartsController.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+      await cart.save();
+      return sendResponse(res, 200, cart, 'Product added to cart');
+    }
+    const findProduct = cartExist.products.find(
+      p => p.productId === product.productId
+    );
+    if (findProduct) {
+      return sendResponse<null>(
+        res,
+        404,
```
Comment (fab-ryan) [inline src/controllers/cartsController.ts]: if product is exist it's doesn't mean it's 404 no we must have special status code

File: src/controllers/cartsController.ts [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  try {
+    const product = req.body as cartProductInterface;
+    const validProduct = await Database.Product.findOne({
+      where: { id: product.productId },
+    });
+    if (!validProduct) {
+      return sendResponse<null>(res, 404, null, 'Product not found');
+    }
+    if (validProduct.quantity < product.quantity) {
+      return sendResponse<null>(res, 404, null, 'Quantity Exceeds Stock');
```
Comment (fab-ryan) [inline src/controllers/cartsController.ts]: if it's exceed it's should be [like](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406)

File: src/controllers/cartsController.ts [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    }
+    const cartExist = await Database.Cart.findOne({ where: { userId: id } });
+    if (!cartExist) {
+      const cart = Database.Cart.build({
+        userId: id,
+        products: [product],
+        totalprice: validProduct.price * product.quantity,
+      });
+      await cart.save();
+      return sendResponse(res, 200, cart, 'Product added to cart');
```
Comment (fab-ryan) [inline src/controllers/cartsController.ts]: if we create we use 201

**Review by fab-ryan — CHANGES_REQUESTED** [review]
status code must be changed

**Review by fab-ryan — CHANGES_REQUESTED** [review]
First figured out why your test are not passing

**Review by Mukunzijames — APPROVED** [review]

**Review by musabehonore — APPROVED** [review]

**Review by Nadinefiona — APPROVED** [review]

**Review by TuyishimireEric — APPROVED** [review]

**Review by fab-ryan — APPROVED** [review]

**Pull request merged**